### PR TITLE
feat: print styling

### PR DESCRIPTION
**Pull request by DilaraGueler**
## PR Type

[X] Feature

## What Is the Current Behavior?

No print rules exist.

## What Is the New Behavior?

General print rules were introduced to the application without a header, and footer, with a reasonable layout, and nice page breaks.

## Does this PR Introduce a Breaking Change?

[X] No

## Other Information

[AB#83954](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/83954)

### CODE REVIEW

File: projects/requisition-management/src/app/pages/requisition-detail/requisition-detail-page.component.html [hunk 1]
```diff
@@ -115,13 +115,13 @@ <h3>{{ 'checkout.order_summary.heading' | translate }}</h3>
     ></ish-requisition-reject-dialog>
 
     <div class="section d-flex d-flex justify-content-between">
-      <a [routerLink]="['../../' + view, { status: requisition.approval.statusCode }]">
+      <a [routerLink]="['../../' + view, { status: requisition.approval.statusCode }]" class="noprint">
```
Comment (andreassteinmann) [inline projects/requisition-management/src/app/pages/requisition-detail/requisition-detail-page.component.html:118]: Please rename the class to `no-print` (and change at all places in HTML an SCSS.

File: src/styles/global/print.scss [hunk 2]
```diff
@@ -1,8 +1,96 @@
...
 // CUSTOMIZED PRINT STYLES
+@import 'variables';
 
 @media print {
+  @page {
+    margin-top: 100px;
+  }
+
+  @supports (-moz-appearance: none) {
+    .logo {
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:11]: Please put the logo styling at one place (not at the top and the bottom of the file).

File: src/styles/global/print.scss [hunk 3]
```diff
@@ -1,8 +1,96 @@
 //
 // CUSTOMIZED PRINT STYLES
+@import 'variables';
 
 @media print {
+  @page {
+    margin-top: 100px;
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:7]: Please consider adding a paper size like A4 that the user is able to select portrait and landscape mode.

File: src/styles/global/print.scss [hunk 4]
```diff
@@ -1,8 +1,96 @@
 //
 // CUSTOMIZED PRINT STYLES
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:2]: Please add a general description like "apply general print styles ... for PDF creation ... apply to pages with a print icon"

File: src/styles/global/print.scss [hunk 5]
```diff
@@ -1,8 +1,96 @@
...
+    .list-item-row {
+      margin-right: 2.5%;
+    }
+  }
+
+  .line-item-list,
+  .order-summary {
+    display: inline-block;
+    break-before: page !important;
+    page-break-before: always !important; /* Older browser versions */
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:88]: change "Older" to "older" (lowercase)

File: src/styles/global/print.scss [hunk 6]
```diff
@@ -1,8 +1,96 @@
...
+    .logo {
+      margin-left: 25px !important;
+    }
+  }
+
   a[href]::after {
     content: none !important;
   }
+
+  button {
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:20]: Please move the button styling down to where the other button styling is applied. You should first have all items listed which are not displayed and afterwards the elements where a certain visual styling is applied. It makes it easier to read the CSS file.

File: src/styles/global/print.scss [hunk 7]
```diff
@@ -1,8 +1,96 @@
...
+  .fa-print,
+  .fa-trash-can,
+  .header-utility,
+  .marketing-area,
+  .navbar-toggler,
+  .product-tile-actions,
+  .share-label,
+  .toast-container,
+  .user-links {
+    display: none !important;
```
Comment (andreassteinmann) [inline src/styles/global/print.scss:55]: Good order of elements!

File: src/styles/global/print.scss (hunk 3 above)
Comment (DilaraGueler) [reply src/styles/global/print.scss:17]: Tried different page size options. It breaks the whole design.

File: src/styles/global/print.scss (hunk 7 above)
Comment (DilaraGueler) [reply src/styles/global/print.scss:57]: Thanks!

**Comment by DilaraGueler** [comment]
> Is there any solution, to make the borders around the green fields not red but green? Because in this case the buyer is still in the green.
> 
> image

Done! The options were green, orange, and red. The borders will be colored according to the background colors.

**Comment by DilaraGueler** [comment]
> The print document name is a little bit misleading. My Account_Intershop-PWA.pdf

Should be asked the PO, if file naming will be in the scope of this task.

File: src/styles/global/print.scss (hunk 3 above)
Comment (andreassteinmann) [reply src/styles/global/print.scss:17]: True, I tried several approaches with the @page size but they always create unwanted layouts even in Chrome.

**Review by andreassteinmann — APPROVED** [review]

**Review by iwiederhold — APPROVED** [review]
I like it! THX

**Pull request merged**